### DISCUSSION TITLE
No init type hints when using declarative tables

### DISCUSSION THREAD

**Discussion by lovetox** [sqlalchemy/sqlalchemy#11033] [Usage Questions] [answered]
Hi,

i tried to define a table declarative

```python3
class Base(DeclarativeBase):
    pass

class Account(Base):
    __tablename__ = "account"

    entitykey: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(unique=True)

Account(wrongarg=1)
```

I dont see any typehints on Account, and when i create that object i can pass everything without error.
im using pyright, and it shows me the type as `**kw: Any`

Is this expected?

**Comment by CaselIT** [comment, marked as answer]
Hi,

Yes, it's the expected behaviour when not using dataclasses

**Comment by me-v2** [comment]
still issuue

**Reply by CaselIT** [reply]
That's impossible to solve with the current capability of python. The best there is using dataclasses to piggy back on their support
